pkg/internal/util/cli/spinner.go

```diff
@@ -60,35 +63,52 @@ var _ io.Writer = &Spinner{}
 // NewSpinner initializes and returns a new Spinner that will write to w
 func NewSpinner(w io.Writer) *Spinner {
 	return &Spinner{
-		stop:   make(chan struct{}, 1),
-		ticker: time.NewTicker(time.Millisecond * 100),
-		mu:     &sync.Mutex{},
-		writer: w,
+		stop:    make(chan struct{}, 1),
+		stopped: make(chan struct{}, 1),
+		mu:      &sync.Mutex{},
+		writer:  w,
 	}
 }
 
 // SetPrefix sets the prefix to print before the spinner
 func (s *Spinner) SetPrefix(prefix string) {
 	s.mu.Lock()
+	defer s.mu.Unlock()
 	s.prefix = prefix
-	s.mu.Unlock()
 }
 
 // SetSuffix sets the suffix to print after the spinner
 func (s *Spinner) SetSuffix(suffix string) {
 	s.mu.Lock()
+	defer s.mu.Unlock()
 	s.suffix = suffix
-	s.mu.Unlock()
 }
 
 // Start starts the spinner running
 func (s *Spinner) Start() {
+	// don't start if we've already started, also flag that we've started
+	if !atomic.CompareAndSwapInt32(&s.running, 0, 1) {
+		return
+	}
+	// start / create a frame ticker
+	func() {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		s.ticker = time.NewTicker(time.Millisecond * 100)
```

Comment:
we also now release the ticker resources, not just when the binary exits. 
this is not a big deal, but nice for library users if they choose to use the cli logging behavior...

this does mean the start / stop restarts the frame cycle, but we do that when we're writing some other line anyhow, so this actually makes sense as an incidental change.

Reply:
we don't stop the ticker often now. actual stop will release this though.